youtube_dl/utils.py

```diff
@@ -842,6 +854,41 @@ def http_response(self, req, resp):
     https_response = http_response
 
 
+def make_socks_conn_class(base_class, socks_proxy):
+    assert issubclass(base_class, (
+        compat_http_client.HTTPConnection, compat_http_client.HTTPSConnection))
+
+    url_components = compat_urlparse.urlparse(socks_proxy)
```

Comment:
This can be fixed by "registering" SOCKS protocols globally in ```utils.py```:
```Python
# "Register" SOCKS protocols
for scheme in ('socks', 'socks4', 'socks5'):
    if scheme not in compat_urlparse.uses_netloc:
        compat_urlparse.uses_netloc.append(scheme)
```
And the result on Python 2.6.2 is: (I didn't run my SOCKS server, so connection refused)
```
$ python2.6 youtube_dl/__main__.py test:youtube --proxy "socks5://127.0.0.1/" -vF
[debug] System config: []
[debug] User config: []
[debug] Command-line args: [u'test:youtube', u'--proxy', u'socks5://127.0.0.1/', u'-vF']
[debug] Encodings: locale UTF-8, fs UTF-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2016.04.19
[debug] Git HEAD: 4953d30
[debug] Python version 2.6.2 - Linux-4.5.1-1-ARCH-x86_64-with-glibc2.2.5
[debug] exe versions: avconv v12_dev0-2591-gd12b5b2, avprobe v12_dev0-2591-gd12b5b2, ffmpeg 3.0.2, ffprobe 3.0.2, rtmpdump 2.4
[debug] Proxy map: {u'http': u'socks5://127.0.0.1/', u'https': u'socks5://127.0.0.1/'}
[TestURL] Test URL: http://www.youtube.com/watch?v=BaW_jenozKc&t=1s&end=9
[youtube] BaW_jenozKc: Downloading webpage
ParseResult(scheme=u'socks5', netloc=u'127.0.0.1', path=u'/', params='', query='', fragment='')
ERROR: Unable to download webpage: <urlopen error [Errno 111] Connection refused> (caused by URLError(error(111, 'Connection refused'),))
  File "/home/yen/Executables/Multimedia/youtube-dl/youtube_dl/extractor/common.py", line 388, in _request_webpage
    return self._downloader.urlopen(url_or_request)
  File "/home/yen/Executables/Multimedia/youtube-dl/youtube_dl/YoutubeDL.py", line 1942, in urlopen
    return self._opener.open(req, timeout=self._socket_timeout)
  File "/usr/lib/python2.6/urllib2.py", line 383, in open
    response = self._open(req, data)
  File "/usr/lib/python2.6/urllib2.py", line 401, in _open
    '_open', req)
  File "/usr/lib/python2.6/urllib2.py", line 361, in _call_chain
    result = func(*args)
  File "/home/yen/Executables/Multimedia/youtube-dl/youtube_dl/utils.py", line 921, in https_open
    req, **kwargs)
  File "/usr/lib/python2.6/urllib2.py", line 1105, in do_open
    raise URLError(err)
```
Is it acceptable?